### PR TITLE
Add stop screen for company type not allowed to verify

### PR DESCRIPTION
**Pull request by hepsimo**
- fix fetchCompany middleware to use companyNumber query param as fallback if submission companyNumber not available
- fix/update corresponding unit and integration test scenarios
- match page title to prototype

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/routers/handlers/stop-screen/stopScreenHandler.ts [hunk 1]
```diff
@@ -73,6 +74,17 @@ const setContent = async (req: Request, stopType: STOP_TYPE, baseViewData: BaseV
                 extraData: [env.GET_RP01_LINK, env.GET_PSC01_LINK, env.POST_TO_CH_LINK, PrefixedUrls.START]
             };
         }
+        case STOP_TYPE.SUPER_SECURE: {
+            return {
+                ...baseViewData,
+                ...getLocaleInfo(locales, lang),
+                templateName: stopType,
+                currentUrl: resolveUrlTemplate(stopScreenPrefixedUrl, stopType),
```
Comment (PhilipGlover) [inline src/routers/handlers/stop-screen/stopScreenHandler.ts:82]: the currentUrl in this Super Secure block has lost the 'addSearchParams' function
`addSearchParams(getUrlWithStopType(stopScreenPrefixedUrl, stopType), { companyNumber, lang })`

**Review by PhilipGlover — COMMENTED** [review]
merge from main needed to resolve conflicts, and I notice the same fix in `fetchCompany` has been done by Al which is in main.

**Review by PhilipGlover — APPROVED** [review]
have run and accessed new screen, which looks good

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Pull request merged**